### PR TITLE
Feat/sl 870 products purchase history [SL-870]

### PR DESCRIPTION
**Pull request by MilosMarkoni**

### CODE REVIEW

File: src/App/Home/HomeContent/Admin/ProductsPurchaseHistory/state/saga.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  } catch (error) {
+    yield put(
+      asyncGetProductsPurchaseHistory.failure(
+        error && error.message ? error.message : 'Cannot fetch product data.',
+      ),
+    );
+  }
+}
+
+export function* exportUsersEarnings$(action: any) {
```
Comment (Kristina2103) [inline src/App/Home/HomeContent/Admin/ProductsPurchaseHistory/state/saga.ts:87]: You forgot to rename this to exportProductPurchaseHistory :)

File: src/components/ProductsPurchaseHistoryTable/index.tsx [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    },
+    pagination: { flexWrap: 'wrap', justifyContent: 'space-between', borderRadius: 5 },
+    spacer: { flex: '0 0 100%' },
+    sacretEmpty: {
+      marginTop: 50,
+    },
+  }),
+);
+
+export type HeaderProps = {
```
Comment (Kristina2103) [inline src/components/ProductsPurchaseHistoryTable/index.tsx:55]: We have HeaderProps in data/repositories/admin/typedefs, maybe you could import it from there? :)

File: src/App/Home/HomeContent/Admin/ProductsPurchaseHistory/state/selectors.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
+import { createSelector } from 'reselect';
+import { AppState } from '../../../../../state/initialState';
+import { isEmpty } from 'lodash';
+
+const getUserEarningsTableState = ({
```
Comment (Kristina2103) [inline src/App/Home/HomeContent/Admin/ProductsPurchaseHistory/state/selectors.tsx:5]: here also getProductPurchaseHistoryState :)

**Review by Kristina2103 — COMMENTED** [review]
👍 😃

**Review by aleksaToljic — APPROVED** [review]

File: src/App/Home/HomeContent/Admin/ProductsPurchaseHistory/state/saga.ts (hunk 1 above)
Comment (MilosMarkoni) [reply src/App/Home/HomeContent/Admin/ProductsPurchaseHistory/state/saga.ts]: Fixed :)

File: src/components/ProductsPurchaseHistoryTable/index.tsx (hunk 2 above)
Comment (MilosMarkoni) [reply src/components/ProductsPurchaseHistoryTable/index.tsx]: Fixed :)

File: src/App/Home/HomeContent/Admin/ProductsPurchaseHistory/state/selectors.tsx (hunk 3 above)
Comment (MilosMarkoni) [reply src/App/Home/HomeContent/Admin/ProductsPurchaseHistory/state/selectors.tsx]: Fixed :)

**Pull request merged**